From bc058235dfd983d775e2d777d3be3539cbf78578 Mon Sep 17 00:00:00 2001
From: Zhang Qiang <qiang.z.zhang@intel.com>
Date: Tue, 22 Jan 2013 15:49:36 -0500
Subject: [PATCH 1/5] Update uid/gid of build root if uid/gid is not match

This is very useful for sharing build root to other developers,
and create a full build root for using by all developer.

Change-Id: I1edd3830a499cff7ab357d17484075e46bd8319b
---
 build |   12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/build b/build
index aa7a217..82b435f 100755
--- a/build
+++ b/build
@@ -2718,13 +2718,11 @@ for SPECFILE in "${SPECFILES[@]}" ; do
 	    chown "$ABUILD_UID:$ABUILD_GID" $BUILD_ROOT/home/abuild
 	else
 	    if ! egrep "^abuild:x?:${ABUILD_UID}:${ABUILD_GID}" >/dev/null <$BUILD_ROOT/etc/passwd ; then
-		echo "abuild user present in the buildroot ($BUILD_ROOT) but uid:gid does not match"
-		echo "buildroot currently using:"
-		egrep "^abuild:" <$BUILD_ROOT/etc/passwd
-		echo "build script attempting to use:"
-		echo "abuild::${ABUILD_UID}:${ABUILD_GID}:..."
-		echo "build aborting"
-		cleanup_and_exit 1
+                sed -i '/^abuild:/d' $BUILD_ROOT/etc/passwd
+                sed -i '/^abuild:/d' $BUILD_ROOT/etc/group
+                echo "abuild:x:${ABUILD_UID}:${ABUILD_GID}:Autobuild:/home/abuild:/bin/bash" >>$BUILD_ROOT/etc/passwd
+                echo "abuild:x:${ABUILD_GID}:" >>$BUILD_ROOT/etc/group
+                chown "$ABUILD_UID:$ABUILD_GID" $BUILD_ROOT/home/abuild -R
 	    fi
 	fi
 	if test -f $BUILD_ROOT/etc/shadow ; then
-- 
1.7.9.5

